src/client/Microsoft.Identity.Client/Internal/Broker/BrokerSilentRequest.cs

```diff
@@ -0,0 +1,102 @@
+    internal class BrokerSilentRequest
+    {
+        internal Dictionary<string, string> BrokerPayload { get; set; } = new Dictionary<string, string>();
+        internal IBroker Broker { get; }
+        private readonly AcquireTokenSilentParameters _silentParameters;
+        private readonly AuthenticationRequestParameters _authenticationRequestParameters;
+        private readonly IServiceBundle _serviceBundle;
+        //private readonly AuthorizationResult _authorizationResult;
+
+        internal BrokerSilentRequest(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenSilentParameters acquireTokenSilentParameters,
+            IServiceBundle serviceBundle,
+            //AuthorizationResult authorizationResult,
+            IBroker broker)
+        {
+            _authenticationRequestParameters = authenticationRequestParameters;
+            _silentParameters = acquireTokenSilentParameters;
+            _serviceBundle = serviceBundle;
+            //_authorizationResult = authorizationResult;
+            Broker = broker;
+        }
+
+        public async Task<MsalTokenResponse> SendTokenRequestToBrokerAsync()
+        {
+            _authenticationRequestParameters.RequestContext.Logger.Info(LogMessages.CanInvokeBrokerAcquireTokenWithBroker);
+
+            return await SendAndVerifyResponseAsync().ConfigureAwait(false);
+        }
+
+        private async Task<MsalTokenResponse> SendAndVerifyResponseAsync()
+        {
+            CreateRequestParametersForBroker();
+
+            MsalTokenResponse msalTokenResponse =
+                await Broker.AcquireTokenUsingBrokerAsync(BrokerPayload).ConfigureAwait(false);
+
+            ValidateResponseFromBroker(msalTokenResponse);
+            return msalTokenResponse;
+        }
+
+        internal void CreateRequestParametersForBroker()
+        {
+            BrokerPayload.Clear();
+            BrokerPayload.Add(BrokerParameter.Authority, _authenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority);
+            string scopes = EnumerableExtensions.AsSingleString(_authenticationRequestParameters.Scope);
+
+            BrokerPayload.Add(BrokerParameter.Scope, scopes);
+            BrokerPayload.Add(BrokerParameter.ClientId, _authenticationRequestParameters.ClientId);
+            BrokerPayload.Add(BrokerParameter.CorrelationId, _authenticationRequestParameters.RequestContext.Logger.CorrelationId.ToString());
+            BrokerPayload.Add(BrokerParameter.ClientVersion, MsalIdHelper.GetMsalVersion());
+            BrokerPayload.Add(BrokerParameter.RedirectUri, _serviceBundle.Config.RedirectUri);
+
+            string extraQP = string.Join("&", _authenticationRequestParameters.ExtraQueryParameters.Select(x => x.Key + "=" + x.Value));
+            BrokerPayload.Add(BrokerParameter.ExtraQp, extraQP);
+
+            BrokerPayload.Add(BrokerParameter.Username, _authenticationRequestParameters.Account?.Username ?? string.Empty);
```

Comment:
what's the difference between silentRequest.Account and _silentParameters.Account?

Reply:
nothing, however, this line is not needed